Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -704,20 +709,22 @@ void winhttp_http_task::callback_status_data_available(
     // Flush response buffer from previous call to WinHttpReadData
     if (pRequestContext->m_responseBuffer.size())
     {
-        flush_response_buffer(pRequestContext);
+        HRESULT hr = flush_response_buffer(pRequestContext);
+        if (FAILED(hr)) {
+            pRequestContext->complete_task(hr);
+            return;
+        }
     }
 
     // Read new data into buffer
     if (newBytesAvailable > 0)
     {
-        size_t oldSize = pRequestContext->m_responseBuffer.size();
-        size_t newSize = oldSize + newBytesAvailable;
-        pRequestContext->m_responseBuffer.resize(newSize);
+        pRequestContext->m_responseBuffer.resize(newBytesAvailable);
```

Comment:
you probably want a try/catch here for the allocation

Reply:
Looks like these will be caught by the calling function: https://github.com/microsoft/libHttpClient/blob/main/Source/HTTP/WinHttp/winhttp_http_task.cpp#L891